### ISSUE TITLE
mail scripts fail since icinga 2.7.0

### ISSUE THREAD

**Issue by paladox** [Icinga/icinga2#5453]
I am running icinga 2.7.0 updated today. My os is debian stretch 9.1.

I am getting this error

Aug  2 16:42:19 gerrit-mysql icinga2[23628]: [2017-08-02 16:39:37 +0000] warning/PluginNotificationTask: Notification command for object 'gerrit-jenkins.wmflabs.org' (PID: 31615, arguments: '/etc/icinga2/scripts/mail-host-notification.sh' '-4' 'gerrit-jenkins.wmflabs.org' '-6' '' '-b' 'paladox' '-c' 'test' '-d' '2017-08-02 16:39:37 +0000' '-l' 'gerrit-jenkins.wmflabs.org' '-n' 'gerrit-jenkins.wmflabs.org' '-o' 'PING OK - Packet loss = 0%, RTA = 1.71 ms' '-r' 'root@localhost' '-s' 'UP' '-t' 'CUSTOM' '-v' 'false') terminated with exit code 1, output: Requirement parameters are missing.
Aug  2 16:42:19 gerrit-mysql icinga2[23628]: Required parameters:
Aug  2 16:42:19 gerrit-mysql icinga2[23628]:   -4 HOSTADDRESS ($address$)
Aug  2 16:42:19 gerrit-mysql icinga2[23628]:   -d LONGDATETIME ($icinga.long_date_time$)
Aug  2 16:42:19 gerrit-mysql icinga2[23628]:   -e SERVICENAME ($service.name$)
Aug  2 16:42:19 gerrit-mysql icinga2[23628]:   -l HOSTNAME ($host.name$)
Aug  2 16:42:19 gerrit-mysql icinga2[23628]:   -n HOSTDISPLAYNAME ($host.display_name$)
Aug  2 16:42:19 gerrit-mysql icinga2[23628]:   -o SERVICEOUTPUT ($service.output$)
Aug  2 16:42:19 gerrit-mysql icinga2[23628]:   -r USEREMAIL ($user.email$)
Aug  2 16:42:19 gerrit-mysql icinga2[23628]:   -s SERVICESTATE ($service.state$)
Aug  2 16:42:19 gerrit-mysql icinga2[23628]:   -t NOTIFICATIONTYPE ($notification.type$)
Aug  2 16:42:19 gerrit-mysql icinga2[23628]:   -u SERVICEDISPLAYNAME ($service.display_name$)
Aug  2 16:42:19 gerrit-mysql icinga2[23628]: Optional parameters:
Aug  2 16:42:19 gerrit-mysql icinga2[23628]:   -6 HOSTADDRESS6 ($address6$)
Aug  2 16:42:19 gerrit-mysql icinga2[23628]:   -b NOTIFICATIONAUTHORNAME ($notification.author$)
Aug  2 16:42:19 gerrit-mysql icinga2[23628]:   -c NOTIFICATIONCOMMENT ($notification.comment$)
Aug  2 16:42:19 gerrit-mysql icinga2[23628]:   -i ICINGAWEB2URL ($notification_icingaweb2url$, Default: unset)
Aug  2 16:42:19 gerrit-mysql icinga2[23628]:   -f MAILFROM ($notification_mailfrom$, requires GNU mailutils (Debian/Ubuntu) or mailx (RHEL/SUSE))
Aug  2 16:42:19 gerrit-mysql icinga2[23628]:   -v ($notification_sendtosyslog$, Default: false)

I noticed that in commands.conf it uses lower case for the vars but in the scripts it's looking for upper case.

**Comment by dnsmichi** [comment]
I assume you just updated the notification scripts, but not the commands.conf and notifications.conf file when asked? All of them changed with v2.7.0 and updates need to be done carefully.

**Comment by paladox** [comment]
Hi, yep i've updated both of those. see https://gerrit.wikimedia.org/r/#/c/369681/4/templates/mail-host-notification.sh.erb and https://gerrit.wikimedia.org/r/#/c/369681/4/templates/commands.conf.erb and https://gerrit.wikimedia.org/r/#/c/369681/4/templates/notifications.conf.erb

**Comment by dnsmichi** [comment]
Hmmm, I would guess that the given service does not have a `display_name` attribute right? Can you show that service from `icinga2 object list` or via REST API call please?

**Comment by paladox** [comment]
Ah sorry, i made a mistake sorry for noise. I copied the service script. My mistake really sorry for that.

**Comment by dnsmichi** [comment]
Ah :) No worries, now I know that Wikimedia upgraded to Icinga 2 :-)

**Comment by paladox** [comment]
Oh the wikimedia did not upgrade to icinga2. I am testing it in wikimedia cloud (the testing side).

**Comment by dnsmichi** [comment]
Oh, still cool that you are doing it :) 

I just double checked the configuration changes in v2.7.0; service.display_name is always set (config compiler sets it to service.name if not specified). So if you omit that on the service object, you're still safe :)

**Comment by paladox** [comment]
Thanks :).
